src/main/java/com/google/javascript/gents/CommentLinkingPass.java

```diff
@@ -176,17 +179,38 @@ private void linkCommentBufferToNode(Node n) {
 
     /** Removes unneeded tags and markers from the comment. */
     private String filterCommentContent(Type type, String comment) {
-      Pattern[] replacements = (type == Type.JSDOC) ? JSDOC_REPLACEMENTS : COMMENT_REPLACEMENTS;
-      for (Pattern p : replacements) {
-        Matcher m = p.matcher(comment);
-        if (m.find() && m.group("keep") != null && m.group("keep").trim().length() > 0) {
-          // keep documentation, if any
-          comment = m.replaceAll("${block}${keep}");
-        } else {
-          // nothing to keep, remove the line
-          comment = m.replaceAll("");
+      if (type == Type.JSDOC) {
+        for (Pattern p : JSDOC_REPLACEMENTS_WITH_KEEP) {
+          Matcher m = p.matcher(comment);
+          if (m.find() && m.group("keep") != null && m.group("keep").trim().length() > 0) {
+            // keep documentation, if any
+            comment = m.replaceAll("${block}${keep}");
+          } else {
+            // nothing to keep, remove the line
+            comment = m.replaceAll("");
+          }
+        }
+
+        for (Pattern p : JSDOC_REPLACEMENTS_NO_KEEP) {
+          Matcher m = p.matcher(comment);
+          boolean hasAMatch = m.find();
+          if (hasAMatch) {
```

Comment:
probably can inline hasAMatch since it is used only once.

Reply:
Done